kernel/tests/read.rs

```diff
@@ -405,7 +410,18 @@ fn read_table_data(path: &str, expected: Vec<&str>) -> Result<(), Box<dyn std::e
 
     let table = Table::new(url);
     let snapshot = table.snapshot(&table_client, None)?;
-    let scan = ScanBuilder::new(snapshot).build();
+    let mut scan_builder = ScanBuilder::new(snapshot.clone());
+
+    if let Some(select_cols) = select_cols {
+        let table_schema = snapshot.schema();
+        let selected_fields: Vec<StructField> = select_cols
+            .iter()
+            .map(|col| table_schema.field(col).cloned().unwrap())
+            .collect();
+        let read_schema = Arc::new(Schema::new(selected_fields));
+        scan_builder = scan_builder.with_schema(read_schema);
```

Comment:
I think this is the third place we have a conditional `with_schema` call, that complicates control flow a bit. Should we add a `with_schema_opt` helper method that streamlines this case for us?
```rust
let read_schema = select_cols.map(|cols| ... );
let scan_builder = ScanBuilder::new(snapshot.clone())
    .with_read_schema_opt(read_schema)
    .build();
```